### PR TITLE
Fix a bug to make failure on reading blob fails account service update

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
When reading blob fails, the account service update should also fail. This pr would fix this bug.

### CODE REVIEW

File: ambry-account/src/test/java/com/github/ambry/account/RouterStoreTest.java [hunk 1]
```diff
@@ -137,6 +135,36 @@ public void testUpdateAndFetch() throws Exception {
...
+    // Now clear the router
+    router.close();
+
+    // generate another new account and test update and fetch on this account
+    Map<Short, Account> anotherIdToRefAccountMap = new HashMap<>();
+    AccountTestUtils.generateRefAccounts(anotherIdToRefAccountMap, idtoRefContainerMap, accountIDSet, 1, 1);
+    for (Map.Entry<Short, Account> entry : anotherIdToRefAccountMap.entrySet()) {
+      idToRefAccountMap.put(entry.getKey(), entry.getValue());
+    }
+    boolean succeeded = store.updateAccounts(anotherIdToRefAccountMap.values());
```
Comment (jsjtzyy) [inline ambry-account/src/test/java/com/github/ambry/account/RouterStoreTest.java]: It seems no need to use `succeeded` boolean variable here. Merge line164 and 165

File: ambry-account/src/main/java/com/github/ambry/account/RouterStore.java [hunk 2]
```diff
@@ -237,6 +237,10 @@ public ZNRecord update(ZNRecord znRecord) {
           // If this is not for backfill, then just read account metadata from blob, otherwise, initialize it with
           // an empty map and fill it up with the accountMap passed to constructor.
           accountMap = (!forBackFill) ? readAccountMetadataFromBlobID(blobIDAndVersion.blobID) : new HashMap<>();
+          if (accountMap == null) {
+            logAndThrowIllegalStateException("Fail to read account metadata from blob id " + blobIDAndVersion.blobID,
```
Comment (jsjtzyy) [inline ambry-account/src/main/java/com/github/ambry/account/RouterStore.java]: why not do this in `readAccountMetadataFromBlobID()` ?

File: ambry-account/src/test/java/com/github/ambry/account/RouterStoreTest.java [hunk 3]
```diff
@@ -137,6 +135,35 @@ public void testUpdateAndFetch() throws Exception {
...
+        new RouterStore(accountServiceMetrics, backup, helixStore, new AtomicReference<>(router), forBackfill,
+            TOTAL_NUMBER_OF_VERSION_TO_KEEP);
+    Map<Short, Account> idToRefAccountMap = new HashMap<>();
+    Map<Short, Map<Short, Container>> idtoRefContainerMap = new HashMap<>();
+    Set<Short> accountIDSet = new HashSet<>();
+    // generate an new account and test update and fetch on this account
+    AccountTestUtils.generateRefAccounts(idToRefAccountMap, idtoRefContainerMap, accountIDSet, 1, 1);
+    assertUpdateAndFetch(store, idToRefAccountMap, idToRefAccountMap, 1, 1);
+
+    // Now clear the router
```
Comment (jsjtzyy) [inline ambry-account/src/test/java/com/github/ambry/account/RouterStoreTest.java]: minor: Now close the router to trigger get blob failure

File: ambry-account/src/main/java/com/github/ambry/account/RouterStore.java (hunk 2 above)
Comment (justinlin-linkedin) [reply ambry-account/src/main/java/com/github/ambry/account/RouterStore.java]: I suppose we could, the readAccountMetadataFromBlobID function is also called by the fetchAccountMetadataFromZNRecord function and that function should not throw any exception. So in the readAccountMetadataFromBlobID, I catch the exception and return null.

**Pull request merged**